tools/decompile.py

```diff
@@ -172,14 +168,14 @@ def _guess_unknown_type(self, decompiled_code: str) -> str:
 
     def _infer_src_path(self) -> Optional[Path]:
         inferred_c_name = f"{self.abspath.parent.parent.name if self.version == "saturn" else self.abspath.parent.name}.c"
-        inferred_c_files = (
-            file
-            for file in self.src_dir.rglob(inferred_c_name)
-            if self.overlay in file.parts or self.overlay in file.name
-        )
-        return next(
-            (path for path in inferred_c_files if self.name in path.read_text()), None
-        )
+        files = [file for file in self.src_dir.rglob(inferred_c_name)]
+        if len(files) > 1 and self.overlay:
+            files = [
+                file
+                for file in files
+                if self.overlay in file.parts or self.overlay in file.name
+            ]
+        return files[0]
```

Comment:
What is the purpose of completely changing this around?
It takes away the function being able to guarantee that it has returned the correct file, as well as adding unnecessary iteration.

Reply:
When `self.overlay` is nil, this logic just crashes. It would be nicer to receive a suggestion how to keep the logic while avoiding unnecessary iterations, as you said.

Reply:
I would have given a suggestion previously, but didn't understand the goal.
Based on how the code was written, there will not be a circumstance where it can get to this point in the code without something being extracted from the path as an overlay value.  If there is an instance where some value wouldn't be extracted as the overlay, then it would throw this message:
`Could not find function RicEntityMariaPowers, are you sure it exists in version pspeu?`